platform/jobs/edxPlatformPythonUnitTestsPr.groovy

```diff
@@ -49,63 +26,130 @@ catch (any) {
     out.println('Exiting with error code 1')
     return 1
 }
-/* Iterate over the job configurations */
-secretMap.each { jobConfigs ->
-
-    Map jobConfig = jobConfigs.getValue()
-
-    /* Test secret contains all necessary keys for this job */
-    assert jobConfig.containsKey('open')
-    assert jobConfig.containsKey('jobName')
-    assert jobConfig.containsKey('subsetJob')
-    assert jobConfig.containsKey('repoName')
-    assert jobConfig.containsKey('coverageJob')
-    assert jobConfig.containsKey('testengUrl')
-    assert jobConfig.containsKey('platformUrl')
-    assert jobConfig.containsKey('testengCredential')
-    assert jobConfig.containsKey('platformCredential')
-    assert jobConfig.containsKey('platformCloneReference')
-    assert jobConfig.containsKey('workerLabel')
-    assert jobConfig.containsKey('whitelistBranchRegex')
-    assert jobConfig.containsKey('context')
-    assert jobConfig.containsKey('triggerPhrase')
-    assert jobConfig.containsKey('defaultTestengBranch')
-    assert ghprbMap.containsKey('admin')
-    assert ghprbMap.containsKey('userWhiteList')
-    assert ghprbMap.containsKey('orgWhiteList')
-
-    buildFlowJob(jobConfig['jobName']) {
-        if (!jobConfig['open'].toBoolean()) {
-            authorization {
-                blocksInheritance(true)
-                permissionAll('edx')
-                permission('hudson.model.Item.Discover', 'anonymous')
-            }
+
+// This script generates a lot of jobs. Here is the breakdown of the configuration options:
+// Map exampleConfig = [ open: true/false if this job should be 'open' (use the default security scheme or not
+//                       jobName: name of the job
+//                       subsetjob: name of subset job run by this job (shard jobs)
+//                       repoName: name of the github repo containing the edx-platform you want to test
+//                       coveragejob: name of the coverage job to run after the unit tests
+//                       workerlabel: label of the worker to run the subset jobs on
+//                       whitelistbranchregex: regex of git branches to build
```

Comment:
I'm actually not sure myself how this works. Maybe we can improve this help text. Is it that PRs with this branch as the merge target (for the PR jobs) and merges to this branch (for the master jobs) will trigger the job?

Reply:
how is this?